### PR TITLE
Introduce MongoTransactionManager

### PR DESCRIPTION
**Pull request by Praveen2112**
## Description
Minor cleanups in Mongo connector which introduces `MongoTransactionManager`  

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

**Review by krvikash — APPROVED** [review]
LGTM. Some comments.

File: plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/ptf/Query.java [hunk 1]
```diff
@@ -124,11 +121,16 @@ public TableFunctionAnalysis analyze(
             if (!collection.equals(collection.toLowerCase(ENGLISH))) {
                 throw new TrinoException(INVALID_FUNCTION_ARGUMENT, "Only lowercase collection name is supported");
             }
-            RemoteTableName remoteTableName = mongoSession.toRemoteSchemaTableName(new SchemaTableName(database, collection));
+            SchemaTableName schemaTableName = new SchemaTableName(database, collection);
+            MongoTableHandle tableHandle = metadata.getTableHandle(session, schemaTableName);
+            if (tableHandle == null) {
+                throw new TableNotFoundException(schemaTableName, format("Table '%s.%s' not found", database, collection), null);
+            }
+            RemoteTableName remoteTableName = metadata.getTableHandle(session, new SchemaTableName(database, collection)).getRemoteTableName();
```
Comment (marcinsbd) [inline plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/ptf/Query.java:129]: nit: I would split this method from this line

**Review by marcinsbd — APPROVED** [review]

**Comment by Praveen2112** [comment]
@marcinsbd / @krvikash Thanks for the review. AC

**Review by ebyhr — APPROVED** [review]

**Comment by Praveen2112** [comment]
@ebyhr Thanks a lot for the review. AC

**Review by marcinsbd — APPROVED** [review]

**Pull request merged**